src/plop.js

```diff
@@ -66,30 +65,46 @@ function run(env) {
 	}
 
 	// set the default base path to the plopfile directory
-	plop = nodePlop(plopfilePath);
-	generators = plop.getGeneratorList();
-	if (!generator) {
-		switch (generators.length) {
-			case 0:
-				console.error(chalk.red('[PLOP] ') + 'No generator found in plopfile');
-				process.exit(1);
-				break;
-
-			case 1:
-				doThePlop(plop.getGenerator(generators[0].name));
-				break;
-
-			default:
-				out.chooseOptionFromList(generators, plop.getWelcomeMessage())
-					.then(function (generatorName) {
-						doThePlop(plop.getGenerator(generatorName));
-					});
-				break;
+	const plop = nodePlop(plopfilePath);
+	const generators = plop.getGeneratorList();
+	const generatorNames = generators.map(function (v) { return v.name; });
```

Comment:
I think we could use an arrow function + destructuring (we don't need to name the variable) for the map: `generators.map(({name}) => name)`

Reply:
Currently, the plop CLI codebase does not make use of a babel transpile step. So, to order to maintain compatibility with node 4.x, I had to do it this way. Eventually, I'd like to just get babel transpilation in place.

Reply:
I didn't notice, now I understand ;-)